Refactor Device management

aravindavk opened the pull request:
Signed-off-by: Aravinda VK <avishwan@redhat.com>

Madhu-1 commented on plugins/device/deviceutils/store-utils.go:
```diff
@@ -1,166 +1,146 @@
 func SetDeviceState(peerID, deviceName, deviceState string) error {
-
-	devices, err := GetDevices(peerID)
+	resp, err := store.Get(context.TODO(), devicePrefix+peerID+"/"+deviceName)
```
can we call `GetDevice(PeerID,DeviceName)` ? to avoid code duplication?

aravindavk replied:
makes sense. will change it

Madhu-1 commented on plugins/device/deviceutils/store-utils.go:
```diff
@@ -1,166 +1,146 @@
 // UpdateDeviceFreeSize updates the actual available size of VG
-func UpdateDeviceFreeSize(peerid, vgname string) error {
-	deviceDetails, err := GetDevices(peerid)
+func UpdateDeviceFreeSize(peerid, device string) error {
```
`peerid` to `peerID`?

Madhu-1 commented on plugins/device/deviceutils/store-utils.go:
```diff
@@ -1,166 +1,146 @@
 
-	peerInfo, err := peer.GetPeer(peerid)
+// UpdateDeviceFreeSizeByVg updates the actual available size of VG
+func UpdateDeviceFreeSizeByVg(peerid, vgname string) error {
```
same here.

Madhu-1 commented on plugins/device/deviceutils/store-utils.go:
```diff
@@ -172,26 +152,20 @@ func IsVgExist(vgname string) bool {
-func getDeviceAvailableSize(peerid, devicePath string) (string, uint64, error) {
-	devices, err := GetDevices(peerid)
+// GetDeviceAvailableSize gets the device size and vgName using device Path
+func GetDeviceAvailableSize(peerid, device string) (uint64, error) {
```
if possible here also :)

Madhu-1 commented on plugins/device/rest.go:
```diff
@@ -41,6 +42,18 @@ func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
+	}
+
+	if err != errors.ErrDeviceNotFound {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, err)
```
this also could be an internal server error right?

aravindavk replied:
ok

Madhu-1 commented on plugins/device/rest.go:
```diff
@@ -93,29 +93,47 @@ func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
 
-	devices, err := deviceutils.GetDevices(peerID)
+	devName := mux.Vars(r)["device"]
+	if devName != "" && !strings.HasPrefix(devName, "/") {
```
do we require prefix check here?

aravindavk replied:
realized that not required, observed that URLs will not have double `/` when it reaches here.

Madhu-1 commented on plugins/device/init.go:
```diff
@@ -27,6 +27,13 @@ func (p *Plugin) RestRoutes() route.Routes {
 			RequestType:  utils.GetTypeString((*deviceapi.AddDeviceReq)(nil)),
 			ResponseType: utils.GetTypeString((*deviceapi.AddDeviceResp)(nil)),
 			HandlerFunc:  deviceAddHandler},
+		route.Route{
```
`endpoints.md`  need to be regenerated. 
* when we can have e2e test and CLI for this API change?

aravindavk replied:
I will add e2e, this URL I added since I added Location header in Add device handler

aravindavk commented:
@Madhu-1 Addressed the review comments. e2e tests not added to `GET /device/{peerid}/{device}` since restclient function is added by the PR #1118 

I will add e2e once #1118 PR merges or we can add test in the same PR.

aravindavk commented:
rebased after edit device merge and added test case for new device info API
